### ISSUE TITLE
Keep calender open or ignore hide event?

### ISSUE THREAD

**Issue by SDCRoman** [dangrossman/daterangepicker#1824]
Hello! I have a working calender, the only thing, which I cant find right now is a way to keep the calender open.

I have a workaround for keeping it open on site load with a simple focus on the daterange input:

`$('input[name="booking_time"]').focus();`

But before I keep experimenting I'd like to ask whether someone has a better solution or whether I missed this feature somewhere in the docu.

**Comment by barryp** [comment]
Yes, I also have a need too a daterangepicker that's always visible - so it doesn't require any user interaction to first show, and doesn't disappear.  I worked around by using a hacked version 1.2 of this library that commented out a few lines in the hide function - but something official would be nice so we can upgrade to newer versions without having to re-hack them.

**Comment by mickeychu** [comment]
> Yes, I also have a need too a daterangepicker that's always visible - so it doesn't require any user interaction to first show, and doesn't disappear. I worked around by using a hacked version 1.2 of this library that commented out a few lines in the hide function - but something official would be nice so we can upgrade to newer versions without having to re-hack them.

Hi @barryp : Could you please share what you commented out to achieve this? I also need the datepicker to be shown always. Thank you!

**Comment by barryp** [comment]
This might look a bit goofy, but I believe it's what does the trick (I tried to untangle this from some other unrelated hacks).  It's basically the first and then last 2 lines of the "hide" function

```
--- daterangepicker-1.2.js      2017-07-25 15:59:03.000000000 -0500
+++ daterangepicker-1.2-report-start.js 2017-07-25 15:59:03.000000000 -0500
@@ -405,16 +405,18 @@
         },
 
         hide: function (e) {
+/*
             this.container.hide();
-
+*/
             if (!this.startDate.isSame(this.oldStartDate) || !this.endDate.isSame(this.oldEndDate))
                 this.notify();
 
             this.oldStartDate = this.startDate.clone();
             this.oldEndDate = this.endDate.clone();
-
+/*
             $(document).off('mousedown', this.hide);
             this.element.trigger('hidden', { picker: this });
+            */
         },
```

**Comment by mickeychu** [comment]
Thank you @barryp  !

**Comment by mercurio-app** [comment]
While a hack like this one works, it would be great if this were introduced in a new version, i guess an options boolean, to tell the hide method if it should hide the calendar after applying the changes or not. Thanks for this library, i wish i had found it earlier!

**Comment by iambhavesh** [comment]
Is there any solution for this issue??

**Comment by montorodiaz** [comment]
I'm going for a basic solution with a style in my css:

        .daterangepicker {
            display: block!important;
        }

When hide, put an inline style: display:none.

If you need the cancel button, you can try with an event in this button.

**Comment by mnort9** [comment]
Here's a similar hack to override the hide function of a daterangepicker instance rather than directly in the library. The override is just commenting out the hide functionality

```
$('#daterange').data('daterangepicker').hide = function() { 
      if (!this.isShowing) return;

      //incomplete date selection, revert to last values
      if (!this.endDate) {
          this.startDate = this.oldStartDate.clone();
          this.endDate = this.oldEndDate.clone();
      }

      //if a new date range was selected, invoke the user callback function
      if (!this.startDate.isSame(this.oldStartDate) || !this.endDate.isSame(this.oldEndDate))
          this.callback(this.startDate.clone(), this.endDate.clone(), this.chosenLabel);

      // //if picker is attached to a text input, update it
      this.updateElement();

      // === Begin Override ===

      // $(document).off('.daterangepicker');
      // $(window).off('.daterangepicker');
      // this.container.hide();
      // this.element.trigger('hide.daterangepicker', this);
      // this.isShowing = false;

      // === End Override ===
    };

```
